src/__tests__/pet.test.js

```diff
@@ -0,0 +1,168 @@
+const { Pet } = require ('../pet.js');
+
+const ERROR_MESSAGE = 'Your pet is no longer alive :(';
+
+let fido;
+beforeEach(() => {
+    fido = new Pet('Fido');
+});
+
+describe ('Pet', () => {
+
+  test('Returns an instance of Pet', () => {
+    expect(fido).toBeInstanceOf(Pet);
+  });
+
+  test('Returns the pets name', () => {
+    expect(fido.name).toBe('Fido');
+  });
+});
+
+describe ('growUp', () => {
+  
+  test ('has an initial age of 0', () => {
+    expect (fido.age).toEqual (0);
+   });
+
+   test ('increments the age by 1', () => {
+    fido.growUp();
+    expect (fido.age).toEqual (1);
+  });
+  
+  test ('has an initial hunger of 0', () => {
+    expect (fido.hunger).toEqual (0);
+  })
+
+  test ('increments the hunger by 5', () => {
+    fido.growUp();
+    expect (fido.hunger).toEqual (5);
+  });
+
+  test ('has an initial fitness of 10', () => {
+    expect (fido.fitness).toEqual (10);
+  });
+
+  test ('decrements the fitness by 3', () => {
+    fido.growUp();
+    expect (fido.fitness).toEqual (7);
+  });
+
+  test(`growUp should throw error: "${ERROR_MESSAGE}"`, () => {
+    fido.age = 30;
+    expect(fido.isAlive).toBe(false);
+    expect(() => {
+        fido.growUp();
+    }).toThrowError(new Error(ERROR_MESSAGE));
+  });
+});
+  
+describe ('walk', () => {
+  test('fitness cannot go above 10', () => {
+    fido.fitness = 10;
+    fido.walk();
+    expect(fido.fitness).toBe(10);
+  });
+
+  test('increments the fitness property by 4', () => {
+    fido.fitness = 5;
+    fido.walk();
+    expect(fido.fitness).toBe(9);
+  });
+
+  test(`walk should throw error: "${ERROR_MESSAGE}"`, () => {
+    fido.fitness = 0;
+    expect(fido.isAlive).toBe(false);
+    expect(() => {
+        fido.walk();
+    }).toThrowError(new Error(ERROR_MESSAGE));
+  });
+});
+   
+describe ('feed', () => {
+
+  test ('hunger cannot go under 0', () => {
+    fido.hunger = 2;
+    fido.feed();
+    expect (fido.hunger).toEqual(0);
+  });
+
+  test ('decrements the hunger by 3', () => {
+    fido.hunger = 4;
+    fido.feed();
+    expect (fido.hunger).toEqual (1);
+  });
+
+  test(`feed should throw error: "${ERROR_MESSAGE}"`, () => {
+    fido.fitness = 0;
```

Comment:
Should this be related to hunger, rather than fitness, when you are testing the feed function.